### PR TITLE
Filters

### PR DESCRIPTION
**Pull request by shapikso**

### CODE REVIEW

File: src/components/Movies/Filters/DoubleRange/DoubleRange.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+// import RangeInput from "./RangeInput";
```
Comment (NataliaHuberna) [inline src/components/Movies/Filters/DoubleRange/DoubleRange.js:2]: comments

File: src/components/Movies/Filters/DoubleRange/DoubleRange.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            <div className="double-range__place">
+                <Range
+                    min={+minValue}
+                    max={+maxValue}
+                    step={step}
+                    defaultValue={[+minValue, +maxValue]}
+                    onChange={(value) => setRange(value[0],value[1])}
+                />
+                {/*<RangeInput maxValue={maxValue} step={step} value={minValue} minValue={minValue}/>*/}
+                {/*<RangeInput maxValue={maxValue} step={step} value={maxValue} minValue={minValue}/>*/}
```
Comment (NataliaHuberna) [inline src/components/Movies/Filters/DoubleRange/DoubleRange.js]: comments

File: src/components/Movies/Filters/Input/Input.scss [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#movie-title {
+  flex: 1 0 auto;
+  //background: url("../../../../filters/search.svg") no-repeat 15px center;
```
Comment (NataliaHuberna) [inline src/components/Movies/Filters/Input/Input.scss]: comments

File: src/components/Movies/Filters/Selector/SelectItem.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+const SelectItem = ({value, title}) => {
+    return (
+        <option value={value}>{title}</option>
+    );
+};
+
+export default SelectItem;
```
Comment (NataliaHuberna) [inline src/components/Movies/Filters/Selector/SelectItem.js:9]: without return

File: src/components/Movies/MovieCard/MovieCard.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import { formatRuntime, formatRate, getRateState } from '../../../helpers/format';
+
+
+const MovieCard = ({ id, backdrop_path, title, runtime, vote_average }) => {
+    const runtimeFormated = formatRuntime(runtime);
+    const formatedRate = formatRate(vote_average);
+    const typeRate = getRateState(vote_average);
+    return (
+        <Link to={`movie/${id}`}>
+            <div className="MovieCard card" data-movieid={id}>
```
Comment (NataliaHuberna) [inline src/components/Movies/MovieCard/MovieCard.js]: data-movied is useless

File: src/components/Movies/MovieCard/MovieCard.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+                    </div>
+                    <div className="film-description">
+                        <div className="film-name">
+                            <span>{title}</span>
+                        </div>
+                        <div className="film-duration">
+                            <span>{runtimeFormated}</span>
+                        </div>
+                    </div>
+                    <div className={"film-rate" + ' ' + `${typeRate}`}>
```
Comment (NataliaHuberna) [inline src/components/Movies/MovieCard/MovieCard.js]: `film-rate  ${typeRate}`

File: src/components/Movies/Movies.js [hunk 7]
```diff
@@ -1,12 +1,136 @@
...
+            }
+        });
+        return result;
+    };
+
+    getCurrentUrl = (page) => {
+        return (this.state.isFiltersSet
+            ? `${this.getUrl()}&page=${page}&per_page=${MOVIE_ON_PAGE}`
+            : `${URL_MOVIE}?page=${page}&per_page=${MOVIE_ON_PAGE}`);
+    };
```
Comment (NataliaHuberna) [inline src/components/Movies/Movies.js]: without return

File: src/components/Movies/Movies.js [hunk 8]
```diff
@@ -1,12 +1,136 @@
...
+        } finally {
+            this.setState({ isLoading: false });
+        }
+    };
+
+    setPage = (page) => this.setState({ currentPage: page });
+
+    loadMore = (page) => {
+        console.log( this.state.currentPage);
+        console.log(page || this.state.currentPage + 1);
```
Comment (NataliaHuberna) [inline src/components/Movies/Movies.js]: console.log

File: src/components/Movies/Movies.js [hunk 9]
```diff
@@ -1,12 +1,136 @@
...
+                        <Loader />
+                    ) : this.state.movies.length ?
+                        (this.state.movies.map((element) => (
+                            <MovieCard
+                                key={element.id}
+                                id={element.id}
+                                backdrop_path={element.backdrop_path}
+                                title={element.title}
+                                runtime = { element.runtime }
+                                vote_average= {element.vote_average}
```
Comment (NataliaHuberna) [inline src/components/Movies/Movies.js]: props in camel case

File: src/components/Movies/Filters/DoubleRange/DoubleRange.js [hunk 10]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+// import RangeInput from "./RangeInput";
+import InputValue from "./InputValue";
+import Slider from "rc-slider";
+import "./slider.scss";
+const { createSliderWithTooltip } = Slider;
+const Range = createSliderWithTooltip(Slider.Range);
+
+const DoubleRange = ({title,minValue,maxValue, step, setRange, minInputValue, maxInputValue}) => {
+    return (
```
Comment (nataliazhuravlova) [inline src/components/Movies/Filters/DoubleRange/DoubleRange.js]: return

File: src/components/Movies/Filters/DoubleRange/InputValue.js [hunk 11]
```diff
@@ -0,0 +1,5 @@
+import React from 'react';
+
+const InputValue = ({value}) => (<input type="text" className="double-range__info-min basic-field" value={value}/>);
```
Comment (nataliazhuravlova) [inline src/components/Movies/Filters/DoubleRange/InputValue.js:3]: you should use custom component

File: src/components/Movies/Filters/DoubleRange/RangeInput.js [hunk 12]
```diff
@@ -0,0 +1,6 @@
+import React from 'react';
+
+const RangeInput = ({minValue,maxValue, step, value}) =>  (
+    <input  type="range" className="double-range__place-item" min={minValue} max={maxValue} step={step} value={value} />);
```
Comment (nataliazhuravlova) [inline src/components/Movies/Filters/DoubleRange/RangeInput.js:4]: you should use custom input

File: src/components/Movies/Filters/DoubleRange/slider.scss [hunk 13]
```diff
@@ -0,0 +1,254 @@
...
+  background-color: #e9e9e9;
+  height: 4px;
+  border-radius: 6px;
+}
+.rc-slider-track {
+  position: absolute;
+  left: 0;
+  height: 4px;
+  border-radius: 6px;
+  /*background-color: #abe2fb;*/
```
Comment (nataliazhuravlova) [inline src/components/Movies/Filters/DoubleRange/slider.scss]: comment

File: src/components/Movies/Filters/DoubleRange/slider.scss [hunk 14]
```diff
@@ -0,0 +1,254 @@
...
+  /*border-color: #96dbfa;*/
+  box-shadow: unset;
+}
+.rc-slider-handle:hover {
+  /*border-color: #57c5f7;*/
+}
+.rc-slider-handle:active {
+  /*border-color: #57c5f7;*/
+  /*box-shadow: 0 0 5px #57c5f7;*/
+  cursor: -webkit-grabbing;
```
Comment (nataliazhuravlova) [inline src/components/Movies/Filters/DoubleRange/slider.scss:49]: delete all comments

File: src/components/Movies/Filters/Filters.js [hunk 15]
```diff
@@ -0,0 +1,95 @@
+import React, {Component} from 'react';
+import './filters.css';
+import Select from "./Selector/Select";
+import {movieLanguages,movieStatus} from '../../../constants/selectors';
```
Comment (nataliazhuravlova) [inline src/components/Movies/Filters/Filters.js]: indents

File: src/components/Movies/Filters/Filters.js [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+};
+
+render() {
+    return (
+        <form id="filters" name="filters" className="form-filters frosted-glass-effect" action="#">
+            <div className="form-filters__header">
+                <FilterInput value={this.props.filters.title} setSearchTitle={this.props.setSearchTitle} />
+                <Select setDelector={this.props.setStatus} options={movieStatus} />
+                <Select setDelector={this.props.setLanguage} options={movieLanguages} />
+            </div>
```
Comment (nataliazhuravlova) [inline src/components/Movies/Filters/Filters.js]: separate comp

File: src/components/Movies/Filters/Filters.js [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
+    this.props.setFilter(false);
+    setTimeout(()=>{
+        this.props.closeModal();
+        this.props.onSubmite();
+        this.setState({isResetDisabled: true});
+    },50);
+};
+
+render() {
+    return (
```
Comment (nataliazhuravlova) [inline src/components/Movies/Filters/Filters.js:34]: destruct props

File: src/components/Movies/Filters/Filters.js [hunk 18]
```diff
@@ -0,0 +1,95 @@
...
+                        placeholderText="Select Date.."
+                        className="basic-field date-picker__input" />
+                </div>
+                <div id="release-date-last" className="form-filters__release-date__item date-picker">
+                    <DatePicker selected={this.props.filters.release_date_last}
+                        onChange={(date) => this.props.setMaxDate(date)}
+                        placeholderText="Select Date.."
+                        className="basic-field date-picker__input" />
+                </div>
+            </div>
```
Comment (nataliazhuravlova) [inline src/components/Movies/Filters/Filters.js]: separate block

File: src/components/Movies/Filters/Filters.js [hunk 19]
```diff
@@ -0,0 +1,95 @@
...
+                    minValue="0"
+                    maxValue="30000"
+                    maxInputValue={this.props.filters.revenue_max}
+                    minInputValue={this.props.filters.revenue_min}
+                />
+            </div>
+            <div className="form-filters__release-date">
+                <div id="release-date-first" className="form-filters__release-date__item date-picker">
+                    <DatePicker selected={this.props.filters.release_date_first}
+                        onChange={(date) => this.props.setMinDate(date)}
```
Comment (nataliazhuravlova) [inline src/components/Movies/Filters/Filters.js]: handler, not func

File: src/components/Movies/Filters/Filters.js [hunk 20]
```diff
@@ -0,0 +1,95 @@
...
+                        placeholderText="Select Date.."
+                        className="basic-field date-picker__input" />
+                </div>
+                <div id="release-date-last" className="form-filters__release-date__item date-picker">
+                    <DatePicker selected={this.props.filters.release_date_last}
+                        onChange={(date) => this.props.setMaxDate(date)}
```
Comment (nataliazhuravlova) [inline src/components/Movies/Filters/Filters.js]: handler, not func

File: src/components/Movies/Filters/Filters.js [hunk 21]
```diff
@@ -0,0 +1,95 @@
...
+                        placeholderText="Select Date.."
+                        className="basic-field date-picker__input" />
+                </div>
+            </div>
+            <div className="form-filters__buttons">
+                <Button onClick={this.formSubmitHandler} className="basic-btn" contentKey ="Submit"/>
+                <Button onClick={this.formResetHandler} type="reset" className="basic-btn"  contentKey="Reset"/>
+                <span onClick={this.props.closeModal} className="close">x</span>
+            </div>
```
Comment (nataliazhuravlova) [inline src/components/Movies/Filters/Filters.js]: separate ButtonsBlock

File: src/components/Movies/Filters/Input/FilterInput.js [hunk 22]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import './Input.scss';
+
+const FilterInput = ({setSearchTitle, value}) => {
+    const changeHandler = (e) => setSearchTitle(e.target.value);
+    return (
+        <input onChange={changeHandler} value={value} type="text" name="movie-title" id="movie-title" placeholder="Search..." className="basic-field" />
```
Comment (nataliazhuravlova) [inline src/components/Movies/Filters/Input/FilterInput.js]: don't create a new input, unnessasary id

File: src/components/Movies/Filters/Selector/Select.js [hunk 23]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import SelectItem from "./SelectItem";
+
+const Select = ({options,setDelector}) => {
+    const clickHandler = (e)=> setDelector(e.target.value);
+    return (
+        <select onChange={clickHandler} name="movie-status" className="basic-field">
+            {options.map((el, index) => <SelectItem key={index} value={el[0]} title={el[1]} />)}
+        </select>
+    );
```
Comment (nataliazhuravlova) [inline src/components/Movies/Filters/Selector/Select.js:10]: make a custom component and use it

File: src/components/Movies/MovieCard/MovieCard.scss [hunk 24]
```diff
@@ -0,0 +1,94 @@
...
+    top: 10px;
+
+    justify-content: center;
+    align-items: center;
+    padding: 4px;
+    border-radius: 5px;
+    text-align: center;
+}
+.good-rate {
+    background-color: #246724;
```
Comment (nataliazhuravlova) [inline src/components/Movies/MovieCard/MovieCard.scss]: in colors

File: src/components/Movies/MovieCard/MovieCard.scss [hunk 25]
```diff
@@ -0,0 +1,94 @@
...
+}
+.bad-rate {
+    background-color: #5B5B5B;
```
Comment (nataliazhuravlova) [inline src/components/Movies/MovieCard/MovieCard.scss]: in colors

File: src/components/Movies/Movies.js [hunk 26]
```diff
@@ -1,12 +1,136 @@
 import React, { Component } from 'react';
+import './movies.scss';
+//import './Movies.scss';
```
Comment (nataliazhuravlova) [inline src/components/Movies/Movies.js]: comment

File: src/components/Movies/Movies.js [hunk 27]
```diff
@@ -1,12 +1,136 @@
...
+    setPopularity = (minPopularity, maxPopularity) => this.setState({filters :{...this.state.filters, popularity_min: minPopularity, popularity_max: maxPopularity}});
+    setVote = (minVote, maxVote) => this.setState({filters :{...this.state.filters,revenue_min: minVote,revenue_max: maxVote}});
+    setLanguage = (language) => this.setState({filters :{...this.state.filters,language: language}});
+    setStatus = (status) => this.setState({filters :{...this.state.filters,status: status}});
+    setSearchTitle = (title) => this.setState({filters :{...this.state.filters,title: title}});
+    setMinDate = (date) => this.setState({filters :{...this.state.filters,release_date_first: date}});
+    setMaxDate = (date) => this.setState({filters :{...this.state.filters,release_date_last: date}});
+
+    toggleFilters = () =>  this.setState({isFiltersHidden: !this.state.isFiltersHidden});
+    setFilter = (isFiltersSet) => this.setState({isFiltersSet: isFiltersSet})
```
Comment (nataliazhuravlova) [inline src/components/Movies/Movies.js]: I don't like this implementation

File: src/components/Movies/MovieCard/MovieCard.js [hunk 28]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { URL_IMG } from '../../../constants/api';
+import { Link } from 'react-router-dom';
+import './MovieCard.scss';
+import { formatRuntime, formatRate, getRateState } from '../../../helpers/format';
+
+
+const MovieCard = ({ id, backdrop_path, title, runtime, vote_average }) => {
```
Comment (nataliazhuravlova) [inline src/components/Movies/MovieCard/MovieCard.js]: where is prop types

File: src/components/common/Loader/Loader.js [hunk 29]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+
+import './Loader.css';
+
+const Loader = () => {
```
Comment (nataliazhuravlova) [inline src/components/common/Loader/Loader.js]: use loader that was already created

File: src/components/common/Button/Button.js [hunk 30]
```diff
@@ -14,11 +14,12 @@ const Button = ({
     if (className) {
         btnClasses.push(className);
     }
+    const handleClick = () => onClick();
```
Comment (nataliazhuravlova) [inline src/components/common/Button/Button.js:17]: for what?

File: src/components/Movies/Movies.js [hunk 31]
```diff
@@ -1,12 +1,136 @@
...
+                        setPopularity={this.setPopularity}
+                        setVote={this.setVote}
+                        setLanguage={this.setLanguage}
+                        setStatus={this.setStatus}
+                        setSearchTitle={this.setSearchTitle}
+                        setMinDate={this.setMinDate}
+                        setMaxDate={this.setMaxDate}
+                        filters={this.state.filters}
+                        setFilter={this.setFilter}
+                        clearFilters={this.clearFilters}
```
Comment (nataliazhuravlova) [inline src/components/Movies/Movies.js]: its very bad

File: src/helpers/format.js [hunk 32]
```diff
@@ -0,0 +1,18 @@
...
+};
+
+export const formatRate = (rate) => (rate ? `${rate}` : NO_RATE);
+
+export const getRateState = (rate) => {
+    if (rate === null) return BAD_RATE;
+    const rateToNumber = Number(rate);
+    if (!Number.isFinite(rateToNumber) || rateToNumber < 7) return BAD_RATE;
+    return GOOD_RATE;
+};
```
Comment (nataliazhuravlova) [inline src/helpers/format.js]: it was in prev pr, check for these func

File: src/components/Movies/Movies.js [hunk 33]
```diff
@@ -1,12 +1,136 @@
...
+        Object.entries(this.state.filters).forEach((element) => {
+            if (element[1] !== '') {
+                result += `&${element[0]}=${element[1].toString().replace(/,/g, '').replace(/ /g, '%20')}`;
+            }
+        });
+        return result;
+    };
+
+    getCurrentUrl = (page) => {
+        return (this.state.isFiltersSet
```
Comment (nataliazhuravlova) [inline src/components/Movies/Movies.js]: this.state.isFiltersSet - unnessary state value. You can simply check if initialFilters values are equal with current ones

File: src/components/Movies/Movies.js [hunk 34]
```diff
@@ -1,12 +1,136 @@
...
+
+    getUrl = () => {
+        let result = `${URL_MOVIE}?`;
+        Object.entries(this.state.filters).forEach((element) => {
+            if (element[1] !== '') {
+                result += `&${element[0]}=${element[1].toString().replace(/,/g, '').replace(/ /g, '%20')}`;
+            }
+        });
+        return result;
+    };
```
Comment (nataliazhuravlova) [inline src/components/Movies/Movies.js]: // Equivalent to `axios.get('https://httpbin.org/get?answer=42')`
const res = await axios.get('https://httpbin.org/get', { params: { answer: 42 } });
Did you try it

**Pull request merged**